task-sdk/src/airflow/sdk/execution_time/supervisor.py

```diff
@@ -845,6 +845,11 @@ def _send_heartbeat_if_needed(self):
         if (time.monotonic() - self._last_heartbeat_attempt) < MIN_HEARTBEAT_INTERVAL:
             return
 
+        if self._terminal_state:
+            # If the task has finished, and we are in "overtime" (running OL listeners etc) we shouldn't
+            # heartbeat
+            return
```

Comment:
nit: lets add a log in here for better trackign?

Reply:
This could happen frequently, I think logging here would be too verbose.


Task logs right now show:

```
[2025-04-09, 15:01:03] INFO - Done. Returned value was: None: source="airflow.task.operators.airflow.providers.standard.decorators.python._PythonDecoratedOperator"
[2025-04-09, 15:01:03] INFO - OpenLineageClient will use `http` transport: source="openlineage.client.client"
[2025-04-09, 15:01:03] INFO - Successfully emitted OpenLineage `COMPLETE` event of id `01961b10-ad80-7b90-bdef-6c9dade57442`: source="airflow.providers.openlineage.plugins.adapter.OpenLineageAdapter"
```

Reply:
Yeah that seems fair.